### PR TITLE
Add documentation for code freeze

### PR DESCRIPTION
**Pull request by raydouglass**
Adds documentation for code freeze and update outdated burn down documentation.

Also adds links from the maintainer page release schedule to the appropriate process descriptions.

### CODE REVIEW

File: releases/process.md [hunk 1]
```diff
@@ -67,18 +70,53 @@ Project Leads
...
-5. Notify project leads process is complete
+3. Create release `M.C` project board
+4. Notify project leads process is complete
+
+Also see the [Burn down guide]({% link resources/burn-down-guide.md %})
+
+
+## Code freeze
+
+Code freeze is the process when the release undergoes thorogh testing. Pull requests are no longer accepted into the development branch. An exception may be made for [hotfix]({% link releases/hotfix.md %}) issues. All pull requests from Burn Down should be merged before Code Freeze begins or be moved to the next release.
```
Comment (ajschmidt8) [inline releases/process.md:89]:
```suggestion
Code freeze is the process when the release undergoes thorough testing. Pull requests are no longer accepted into the development branch. An exception may be made for [hotfix]({% link releases/hotfix.md %}) issues. All pull requests from Burn Down should be merged before Code Freeze begins or be moved to the next release.
```

**Review by ajschmidt8 — APPROVED** [review]
found a typo. otherwise, LGTM

File: releases/process.md [hunk 2]
```diff
@@ -40,25 +40,28 @@ From this point forward you can follow the git branching & release model used by
 Both major and minor releases follow the same steps with little variation. For both, release types there are two key dates that need to be known well in advance:
 
 * `Burn down date`
+* `Code freeze data`
```
Comment (mike-wendt) [inline releases/process.md:43]:
```suggestion
* `Code freeze date`
```

**Comment by mike-wendt** [comment]
Will review the preview shortly

**Review by mike-wendt — APPROVED** [review]

**Comment by mike-wendt** [comment]
Thanks @raydouglass

**Pull request merged**